src/reference/buffer.rs

```diff
@@ -235,7 +235,7 @@ impl IriRefBuf {
 			if let Some(query_len) = self.p.query_len {
 				self.replace(offset..(offset + query_len), new_query.as_ref());
 			} else {
-				self.replace(offset..offset, &[0x3f]);
+				self.replace(offset..offset, b":");
```

Comment:
This is the wrong character. It should be `?` instead of `:`. Fixed in version 2.0.1.